azext_iot/tests/iothub/message_endpoint/test_iothub_message_endpoint_int.py

```diff
@@ -0,0 +1,906 @@
+        batch_frequency_in_seconds=60,
+        encoding="json",
+        file_name_format=custom_file_format,
+        max_chunk_size_in_bytes=10
+    )
+
+    endpoint_output = cli.invoke(
+        "iot hub message-endpoint show -n {} -g {} --en {}".format(
+            iot_hub, iot_rg, endpoint_names[1]
+        )
+    ).as_json()
+
+    assert_endpoint_properties(endpoint_output, expected_sys_endpoint)
+
+    # Use user identity
+    cli.invoke(
+        "iot hub message-endpoint create storage-container -n {} -g {} --en {} --erg {} --endpoint-uri {} --container {} "
+        "--identity {} --auth-type identityBased -b {} -w {}".format(
+            iot_hub,
+            iot_rg,
+            endpoint_names[2],
+            iot_rg,
+            endpoint_uri,
+            container_name,
+            user_id,
+            720,
+            500
+        )
+    )
+
+    expected_user_endpoint = build_expected_endpoint(
+        endpoint_names[2],
+        iot_rg,
+        iot_sub,
+        container_name=container_name,
+        authentication_type=AuthenticationType.IdentityBased.value,
+        endpoint_uri=endpoint_uri,
+        identity=user_id,
+        batch_frequency_in_seconds=720,
+        encoding="avro",
+        file_name_format=default_file_format,
+        max_chunk_size_in_bytes=500
+    )
+
+    endpoint_output = cli.invoke(
+        "iot hub message-endpoint show -n {} -g {} --en {}".format(
+            iot_hub, iot_rg, endpoint_names[2]
+        )
+    ).as_json()
+
+    assert_endpoint_properties(endpoint_output, expected_user_endpoint)
+
+    endpoint_list = cli.invoke(
+        "iot hub message-endpoint list -n {} -g {}".format(
+            iot_hub, iot_rg
+        )
+    ).as_json()
+
+    storage_list = cli.invoke(
+        "iot hub message-endpoint list -n {} -g {} -t {}".format(
+            iot_hub, iot_rg, "storage-container"
+        )
+    ).as_json()
+
+    assert len(storage_list) == 3
+    assert endpoint_list["storageContainers"] == storage_list
+
+    # Delete one event hub endpoint
+    cli.invoke(
+        "iot hub message-endpoint delete -n {} -g {} --en {} -y".format(
+            iot_hub, iot_rg, endpoint_names[0]
+        )
+    )
+
+    # ensure that only one got deleted
+    storage_list = cli.invoke(
+        "iot hub message-endpoint list -n {} -g {} -t {}".format(
+            iot_hub, iot_rg, "storage-container"
+        )
+    ).as_json()
+
+    assert len(storage_list) == 2
+
+    # Delete all event hub endpoints
+    cli.invoke(
+        "iot hub message-endpoint delete -n {} -g {} -t {} -y".format(
+            iot_hub, iot_rg, "storage-container"
+        )
+    )
+
+    endpoint_list = cli.invoke(
+        "iot hub message-endpoint list -n {} -g {} -t {}".format(
+            iot_hub, iot_rg, "storage-container"
+        )
+    ).as_json()
+
+    assert endpoint_list == []
+
+
+def test_iot_cosmos_endpoint_lifecycle(provisioned_cosmosdb_with_identity_module):
```

Comment:
add skipif to make sure we do not test it on an azure cli core version that does not support this